packages/rich-text/src/component/use-input-and-selection.js

```diff
@@ -275,29 +275,10 @@ export function useInputAndSelection( props ) {
 				return;
 			}
 
-			if ( ! isSelected ) {
-				// We know for certain that on focus, the old selection is invalid.
-				// It will be recalculated on the next mouseup, keyup, or touchend
-				// event.
-				const index = undefined;
-
-				record.current = {
-					...record.current,
-					start: index,
-					end: index,
-					activeFormats: EMPTY_ACTIVE_FORMATS,
-				};
-				onSelectionChange( index, index );
-			} else {
+			if ( isSelected ) {
 				applyRecord( record.current );
 				onSelectionChange( record.current.start, record.current.end );
 			}
```

Comment:
It turns out, this is the only "useful" code here that doesn't duplicate `selectionchange` event. It's useful when applying formats (like links), that way the selection is restored at the right position after inserting links. Maybe we could even move this code elsewhere (when we actually update record.current) and avoid the event handler entirely but I'm not familiar enough with this code to try it.